<h1>B09705039_劉惟恩_HW01

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats
import statsmodels.stats.api as sms
from matplotlib import pyplot as plt
%matplotlib inline
# 設定圖形大小; DPI越大圖越大
plt.rcParams["figure.dpi"] = 130

<h2>13-1

<h2>Problem 13.34

檢查兩個母體的變異數是否相同<br>

令 Fixed salary組 的均數與變異數為 $\mu_1$與$\sigma_1^2$。<br>
   Commission組 的均數與變異數為 $\mu_2$與$\sigma_2^2$。

Null Hypothesis: $\sigma_1^2 / \sigma_2^2 = 1$ <br>
Alternative Hypothesis: $\sigma_1^2 / \sigma_2^2 \ne 1$

使用雙尾F檢定，Alpha = 0.05

In [2]:
# Reading dataset
# 讀取資料集
df1 = pd.read_excel('Xr13-34.xlsx')

# 簡單看一下資料集的樣子
# 如果這是你第一次使用這個資料集，使用Excel開啟資料集，並確認程式讀到的跟你在Excel中看到的一樣

print('Head of dataset:')
display(df1.head())

print("Tail of dataset:")
display(df1.tail())

Head of dataset:


Fixed  Commission
0  75558       48705
1  73974       51927
2  68985       56733
3  60239       52997
4  67320       60655

Tail of dataset:


Fixed  Commission
85  49302       58990
86  47334       64509
87  37480       66350
88  61667       56519
89  37098       60039

In [3]:
print("The shape of the dataset is", df1.shape)
df_num = df1.shape[0] - 1
df_denom = df_num

print("degree of freedom for numerator = ", df_num)
print("degree of freedom for denominator = ", df_denom)

The shape of the dataset is (90, 2)
degree of freedom for numerator =  89
degree of freedom for denominator =  89


In [4]:
# 使用.values取用numpy array
varall = df1.var().values
# type(varall)

f_value = varall[0] / varall[1]
print("f_value = ", f_value)

# Alternative Hypothesis: two sided
ptmp = stats.f.cdf(f_value, df_num, df_denom)
p_value = (1-ptmp) * 2
print(f"p_value = {p_value:.4f}")

f_value =  0.954140500125618
p_value = 1.1748


The p-value for equal-variance test is 1.1748, which is larger than the required 0.05 significance level, there is no sufficient evidence to reject the null hypothesis. As a result, we assume that the two populations have equal variance and apply equal-variance t-test on $\mu_1 - \mu_2 = 0$.

Null Hypothesis $\mu_1 - \mu_2 = 0$ <br>
Alternative Hypothesis: $\mu_1 - \mu_2 < 0$.

In [5]:
# t-test using sms.CompareMeans
cm = sms.CompareMeans(sms.DescrStatsW(df1["Fixed"].values), sms.DescrStatsW(df1["Commission"].values))
# usevarstr, ‘pooled’ or ‘unequal’
cm.summary(usevar='pooled', alpha=0.05)

<class 'statsmodels.iolib.table.SimpleTable'>

Here $P > |t|$ represent the two-tailed p-value. In our case, we are performing a left-tailed hypothesis test. The p-value should be $0.038 / 2 = 0.019$. 
The p-value is smaller than the required significance level 0.05. As a result, we reject the null hypothesis in favor of the alternative hypothesis. 

Since the null hypothesis that both groups have equal performance is rejected, we favor the alternative hypothesis and conclude that the commission salesperson outperforms the fixed-salary salesperson.

<h2>Problem 13.38

檢查兩個母體的變異數是否相同<br>

令 Big Bucket組 的均數與變異數為 $\mu_1$與$\sigma_1^2$。<br>
   Medium Bucket組 的均數與變異數為 $\mu_2$與$\sigma_2^2$。

Null Hypothesis: $\sigma_1^2 / \sigma_2^2 = 1$ <br>
Alternative Hypothesis: $\sigma_1^2 / \sigma_2^2 \ne 1$

使用雙尾F檢定，Alpha = 0.05

In [6]:
# Reading dataset
# 讀取資料集
df1 = pd.read_excel('Xr13-38.xlsx')

# 簡單看一下資料集的樣子
# 如果這是你第一次使用這個資料集，使用Excel開啟資料集，並確認程式讀到的跟你在Excel中看到的一樣

print('Head of dataset:')
display(df1.head())

print("Tail of dataset:")
display(df1.tail())

Head of dataset:


Big Bucket  Medium Bucket
0         133             50
1         106             63
2         104             56
3          85             37
4          63             63

Tail of dataset:


Big Bucket  Medium Bucket
43          86             51
44          96             59
45         116             66
46          93             73
47          85             70

In [7]:
print("The shape of the dataset is", df1.shape)
df_num = df1.shape[0] - 1
df_denom = df_num

print("degree of freedom for numerator = ", df_num)
print("degree of freedom for denominator = ", df_denom)

The shape of the dataset is (48, 2)
degree of freedom for numerator =  47
degree of freedom for denominator =  47


In [8]:
# 使用.values取用numpy array
varall = df1.var().values
# type(varall)

f_value = varall[0] / varall[1]
print("f_value = ", f_value)

# Alternative Hypothesis: two sided
ptmp = stats.f.cdf(f_value, df_num, df_denom)
p_value = (1-ptmp) * 2
print(f"p_value = {p_value:.4f}")

f_value =  3.1431291390728475
p_value = 0.0001


The p-value for equal-variance test is 0.0001, which is smaller than the required 0.05 significance level, there is sufficient evidence to reject the null hypothesis. As a result, we assume that the two populations have unequal variance and apply unequal-variance t-test on $\mu_1 - \mu_2 = 0$.

Null Hypothesis $\mu_1 - \mu_2 = 0$ <br>
Alternative Hypothesis: $\mu_1 - \mu_2 > 0$.

In [9]:
# t-test using sms.CompareMeans
cm = sms.CompareMeans(sms.DescrStatsW(df1["Big Bucket"].values), sms.DescrStatsW(df1["Medium Bucket"].values))
# usevarstr, ‘pooled’ or ‘unequal’
cm.summary(usevar='unequal', alpha=0.05)

<class 'statsmodels.iolib.table.SimpleTable'>

Here $P > |t|$ represent the two-tailed p-value. In our case, we are performing a right-tailed hypothesis test. The p-value should be < 0.05. 
The p-value is smaller than the required significance level 0.05. As a result, we reject the null hypothesis in favor of the alternative hypothesis. 

Since the null hypothesis that both groups eat equal amounts is rejected, we favor the alternative hypothesis and conclude that the larger the bucket the more people will eat.

<h2>13-4

<h2>Problem 13.114

檢查兩個母體的變異數是否相同<br>

令 Machine 1 的變異數為 $\sigma_1^2$。<br>
   Machine 2 的變異數為 $\sigma_2^2$。

Null Hypothesis: $\sigma_1^2 / \sigma_2^2 = 1$ <br>
Alternative Hypothesis: $\sigma_1^2 / \sigma_2^2 \ne 1$

使用雙尾F檢定，Alpha = 0.05

In [10]:
# Reading dataset
# 讀取資料集
df1 = pd.read_excel('Xr13-114.xlsx')

# 簡單看一下資料集的樣子
# 如果這是你第一次使用這個資料集，使用Excel開啟資料集，並確認程式讀到的跟你在Excel中看到的一樣

print('Head of dataset:')
display(df1.head())

print("Tail of dataset:")
display(df1.tail())

Head of dataset:


Machine 1  Machine 2
0      0.998      1.003
1      0.997      1.004
2      1.003      0.997
3      1.000      0.996
4      0.999      0.999

Tail of dataset:


Machine 1  Machine 2
6       0.998      1.000
7       1.003      1.005
8       1.004      1.002
9       1.000      1.004
10        NaN      0.996

In [11]:
# 轉換成numpy array
Machine_1 = df1['Machine 1'].values
# 去除NAN
Machine_1 = Machine_1[~np.isnan(Machine_1)]
print("Machine_1 = \n", Machine_1)

Machine_2 = df1['Machine 2'].values
print("Machine_2 = \n", Machine_2)

Machine_1 = 
 [0.998 0.997 1.003 1.    0.999 1.    0.998 1.003 1.004 1.   ]
Machine_2 = 
 [1.003 1.004 0.997 0.996 0.999 1.003 1.    1.005 1.002 1.004 0.996]


In [12]:
print("The shape of the dataset is", df1.shape)
df_num = Machine_1.shape[0] - 1
df_denom = Machine_2.shape[0] - 1

print("degree of freedom for numerator = ", df_num)
print("degree of freedom for denominator = ", df_denom)

The shape of the dataset is (11, 2)
degree of freedom for numerator =  9
degree of freedom for denominator =  10


In [13]:
f_value = stats.describe(Machine_1).variance / stats.describe(Machine_2).variance
print("f_value = ", f_value)

# Alternative Hypothesis: two sided
ptmp = stats.f.cdf(f_value, df_num, df_denom)
p_value = (1-ptmp) * 2
print(f"p_value = {p_value:.4f}")

f_value =  0.5045333333333296
p_value = 1.6821


The p-value for equal-variance test is 1.6821, which is greater than the required 0.05 significance level, there is no sufficient evidence to reject the null hypothesis. As a result, we assume that the two populations have equal variance.

Thus, the manager can't infer that the two machines differ in their consistency of fills.

<h2>Problem 13.117

檢查兩個母體的變異數是否相同<br>

令 Week 1 的變異數為 $\sigma_1^2$。<br>
   Week 2 的變異數為 $\sigma_2^2$。

Null Hypothesis: $\sigma_1^2 / \sigma_2^2 = 1$ <br>
Alternative Hypothesis: $\sigma_1^2 / \sigma_2^2 > 1$

使用F檢定，Alpha = 0.05

In [14]:
# Reading dataset
# 讀取資料集
df1 = pd.read_excel('Xr13-117.xlsx')

# 簡單看一下資料集的樣子
# 如果這是你第一次使用這個資料集，使用Excel開啟資料集，並確認程式讀到的跟你在Excel中看到的一樣

print('Head of dataset:')
display(df1.head())

print("Tail of dataset:")
display(df1.tail())

Head of dataset:


Week 1  Week 2
0      69      75
1      77      69
2      74      73
3      67      72
4      81      71

Tail of dataset:


Week 1  Week 2
95      67      76
96      84      69
97      71      68
98      73      74
99      73      64

In [15]:
print("The shape of the dataset is", df1.shape)
df_num = df1.shape[0] - 1
df_denom = df_num

print("degree of freedom for numerator = ", df_num)
print("degree of freedom for denominator = ", df_denom)

The shape of the dataset is (100, 2)
degree of freedom for numerator =  99
degree of freedom for denominator =  99


In [16]:
# 使用.values取用numpy array
varall = df1.var().values
# type(varall)

f_value = varall[0] / varall[1]
print("f_value = ", f_value)

# Alternative hypothesis: signa1^2 / sigma2^2 > 1
p_value = 1 - stats.f.cdf(f_value, df_num, df_denom)
print(f"p_value = {p_value:.4f}")

f_value =  1.5260582820865725
p_value = 0.0183


The p-value for variance test is 0.0183, which is smaller than the required 0.05 significance level, there is sufficient evidence to reject the null hypothesis in favor of the alternative hypothesis.
As a result, we assume that the second variance is smaller that the first variance.

Thus, we can infer that limiting the minimum and maximum speeds reduces the variation in speeds.